### PR TITLE
doc: lib.lists migrate to doc-comments

### PR DESCRIPTION
**Pull request by hsjobeki**
## Description of changes

Migrate the comments to be doc-comments compliant to rfc145.

@DanielSidhion 

**Doc-comment migration**

- [x] Inner/outer format migrated `/* {nixdoc} */` -> `/** {commonmark} */`
- [x] Clean up function arguments. Please manually remove the redundant line comments. We decided the codemod won't delete anything.
- [x] Reify examples. After the migration is done all examples should be placed at the end under an `# Examples` heading.
    - [ ] Every isolated example should have seperate code fences \`\`\`nix
    - [x] Examples should be surrunded by :::{.example} such that they are properly styled.
    - [x] Make sure examples have a descriptive heading. e.g. `## lib.asserts.assertMsg usage examples`

I'll clean up the manually needed steps once we are sure the automatic migration worked out here.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by hsjobeki** [comment]
https://github.com/nix-community/nixdoc/pull/109 should be merged and updated in nixpkgs before this can be merged.

**Comment by infinisil** [comment]
Just merged that PR and released 3.0.2 with it. Should get an automated PR to update it soon.

**Comment by infinisil** [comment]
https://github.com/NixOS/nixpkgs/pull/295200

File: lib/lists.nix [hunk 1]
```diff
@@ -406,50 +842,113 @@ rec {
...
+
+    :::
+  */
+  partition = builtins.partition;
+
+  /**
+    Splits the elements of a list into many lists, using the return value of a predicate.
+        Predicate should return a string which becomes keys of attrset `groupBy` returns.
+
+        `groupBy'` allows to customise the combining function and initial value
```
Comment (DanielSidhion) [inline lib/lists.nix:907]: Too much indentation added to subsequent lines - this causes text to be rendered as if it were code.

File: lib/lists.nix [hunk 2]
```diff
@@ -481,48 +1007,101 @@ rec {
...
+    zipLists [ 1 2 ] [ "a" "b" ]
+    => [ { fst = 1; snd = "a"; } { fst = 2; snd = "b"; } ]
+    ```
 
-     Example:
-       zipLists [ 1 2 ] [ "a" "b" ]
-       => [ { fst = 1; snd = "a"; } { fst = 2; snd = "b"; } ]
+    :::
   */
   zipLists = zipListsWith (fst: snd: { inherit fst snd; });
```
Comment (DanielSidhion) [inline lib/lists.nix:1031]: No inputs were added because of the currying

**Comment by DanielSidhion** [comment]
@hsjobeki I went through the PR already fixing issues I found. Please double-check my changes and let me know if everything's good, and I'll merge the PR.

**Comment by hsjobeki** [comment]
Very nice thanks for the help. I realized that both `subtractLists` and `intersectLists` have a headline like "subtract list `e` from another list". 
I think it could be nice to either call the inputs `e` and `other list` (im not a fan of that)  or just refer to the name of the inputs. (If they have one)

**Review by DanielSidhion — APPROVED** [review]

**Pull request merged**